core/unix/module_elf.c

```diff
@@ -209,15 +211,38 @@ elf_dt_abs_addr(ELF_DYNAMIC_ENTRY_TYPE *dyn, app_pc base, size_t size, size_t vi
     return tgt;
 }
 
-Elf64_Off module_get_dynstr_offset(app_pc base, size_t view_size)
+/* TODO: Is there a way of accessing elf_loader_t's load_base and image_size here?
+ * If so, then this function may not be needed.
+ */
+static size_t
+module_map_file(os_module_data_t *os_mod_data)
+{
+    int fd;
+    uint64 size;
+
+    fd = open_syscall(os_mod_data->source_file_name, O_RDONLY, 0);
+    ASSERT(fd != -1);
+
+    if (!os_get_file_size_by_handle(fd, &size))
+        ASSERT(false);
+
+    os_mod_data->source_file_map = (byte *)mmap_syscall(NULL, size, PROT_READ,
```

Comment:
Where is the best place to munmap() the files mmap()ed here? The debug build didn't throw up any leaks even though its not been unmaped!

Reply:
This is using a raw syscall to mmap so it cannot be tracked by either the vmm manager or the heap allocator.  Maybe it should use one of the managed calls.

Let me look at the full code before seeing if I have a suggestion for unmapping.